flat_mutation_reader.cc

```diff
@@ -909,6 +909,96 @@ make_flat_mutation_reader_from_fragments(schema_ptr schema, reader_permit permit
     return make_flat_mutation_reader_from_fragments(std::move(schema), std::move(permit), std::move(fragments), pr);
 }
 
+flat_mutation_reader
+make_filtering_reader(flat_mutation_reader rd, const dht::partition_range& pr, const query::partition_slice& slice) {
```

Comment:
You could run_mutation_source_tests() on it (overlayed over memtable reader, for example, which doesn't do filtering).